src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java

```diff
@@ -205,24 +203,15 @@ private String importDemoData(AdminHomePageData pageData)
         }
         
         //produce searchable documents
+        //this is not added to taskQueue because it might throw EntityDoesNotExistException
         List<CommentAttributes> comments = logic.getCommentsForGiver(courseId, pageData.instructorEmail);
```

Comment:
The fact that this can freely throw `EntityDoesNotExistException` is troubling, to say the least. This means if this Exception is thrown, all operations past this line is not executed after all.
It should be wrapped in a try-catch block and a severe log, something like:
```java
List<CommentAttributes> comments = new ArrayList<CommentAttributes>();
try {
    comments = logic.getCommentsForGiver(courseId, pageData.instructorEmail);
} catch (EntityDoesNotExistException e) {
    log.severe(...); // write an appropriate log message
}
```
then it can eventually be moved to the task queue action. Many operations in the code base are guilty of this as well.